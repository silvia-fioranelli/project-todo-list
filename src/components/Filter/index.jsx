import { func } from 'prop-types';

const Filter = (props) => {
  const { setCategory } = props;
  return (
    <div>
      <button onClick={() => setCategory('')}>ALL</button>
      <button onClick={() => setCategory('pending')}>Pending</button>
      <button onClick={() => setCategory('completed')}>Completed</button>
    </div>
  );
};

Filter.propTypes = {
  setCategory: func.isRequired,
};

export default Filter;
