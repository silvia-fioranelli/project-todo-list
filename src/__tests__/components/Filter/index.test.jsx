import { shallow } from 'enzyme';
import Filter from '../../../components/Filter';

describe('Filter', () => {
  describe('rendering', () => {
    test('should render', () => {
      const props = {
        setCategory: jest.fn(),
      };
      const wrapper = shallow(<Filter {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('behaviour', () => {
    test('should reset filter type', () => {
      const props = {
        setCategory: jest.fn(),
      };
      const wrapper = shallow(<Filter {...props} />);
      wrapper.find('button').at(0).simulate('click');
      expect(props.setCategory).toBeCalledTimes(1);
      expect(props.setCategory).toBeCalledWith('');
    });

    test('should set filter type with pending', () => {
      const props = {
        setCategory: jest.fn(),
      };
      const wrapper = shallow(<Filter {...props} />);
      wrapper.find('button').at(1).simulate('click');
      expect(props.setCategory).toBeCalledTimes(1);
      expect(props.setCategory).toBeCalledWith('pending');
    });

    test('should set filter type with completed', () => {
      const props = {
        setCategory: jest.fn(),
      };
      const wrapper = shallow(<Filter {...props} />);
      wrapper.find('button').at(2).simulate('click');
      expect(props.setCategory).toBeCalledTimes(1);
      expect(props.setCategory).toBeCalledWith('completed');
    });
  });
});
