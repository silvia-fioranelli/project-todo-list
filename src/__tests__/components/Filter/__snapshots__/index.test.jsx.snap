// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter rendering should render 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    ALL
  </button>
  <button
    onClick={[Function]}
  >
    Pending
  </button>
  <button
    onClick={[Function]}
  >
    Completed
  </button>
</div>
`;
